### PR TITLE
Update sinon to 17 version

### PR DESCRIPTION
**Pull request by Alyar666**

### CODE REVIEW

File: packages/devextreme/testing/helpers/scheduler/helpers.js [hunk 1]
```diff
@@ -270,10 +271,8 @@ class Appointment extends ClickElementWrapper {
     }
 
     click() {
-        const clock = sinon.useFakeTimers();
         this.getElement().trigger('dxclick');
-        clock.tick(300);
-        clock.restore();
```
Comment (andrewmakarov) [inline packages/devextreme/testing/helpers/scheduler/helpers.js:276]: Why remove **restore** method?

**Review by pomahtri — DISMISSED** [review]

**Review by alexanderPolosatov — DISMISSED** [review]

File: packages/devextreme/testing/helpers/scheduler/helpers.js [hunk 2]
```diff
@@ -270,10 +271,8 @@ class Appointment extends ClickElementWrapper {
...
+        this.clock && this.clock.tick(300);
```
Comment (williamvinogradov) [inline packages/devextreme/testing/helpers/scheduler/helpers.js:275]: Minor
```suggestion
        this.clock?.tick(300);
```

File: packages/devextreme/testing/helpers/scheduler/helpers.js [hunk 3]
```diff
@@ -500,12 +500,9 @@ export class SchedulerTestWrapper extends ElementWrapper {
                 if(isAsync) {
                     click();
                 } else {
-                    const clock = sinon.useFakeTimers();
-
                     click();
 
-                    clock.tick(300);
-                    clock.restore();
+                    this.clock && this.clock.tick(300);
```
Comment (williamvinogradov) [inline packages/devextreme/testing/helpers/scheduler/helpers.js:505]: Minor
```suggestion
                    this.clock?.tick(300);
```

File: packages/devextreme/testing/helpers/vizMocks.js [hunk 4]
```diff
@@ -327,6 +327,17 @@
...
+            const stub = this.stub(name);
+
+            if(stub) {
+                if(stub.resetHistory) {
+                    stub.resetHistory();
+                } else {
+                    stub.reset();
+                }
+            }
+        };
```
Comment (williamvinogradov) [inline packages/devextreme/testing/helpers/vizMocks.js:340]: Minor
```suggestion
        stubPrototype.resetStub = function(name) {
            const stub = this.stub(name);
            
            if(stub?.resetHistory) {
              stub.resetHistory();
              return;
            }
            
            stub?.reset();
        };
```

File: packages/devextreme/testing/helpers/scheduler/helpers.js (hunk 1 above)
Comment (williamvinogradov) [reply packages/devextreme/testing/helpers/scheduler/helpers.js:276]: Looks like that now we use the one ```clock``` instance across the test case
So, it should restore at the end of the test I guess

@Alyar666, am I right? :)

File: packages/devextreme/testing/tests/DevExpress.ui.widgets.scheduler/dataSource.tests.js [hunk 5]
```diff
@@ -760,46 +760,52 @@ module('ArraySore(auto generated id)', moduleConfig, () => {
...
-            onAppointmentDeleted: e => {
-                assert.deepEqual(e.appointmentData, {
-                    id: 'bc0eadf8-608a-491f-9f23-e5a8100352e7',
-                    startDate: new Date(2020, 6, 10, 1),
-                    endDate: new Date(2020, 6, 10, 2),
-                    text: 'Test'
-                }, 'Appointment should be equal with dataSource appointment on onAppointmentDeleted event');
-            }
-        });
+        try {
```
Comment (williamvinogradov) [inline packages/devextreme/testing/tests/DevExpress.ui.widgets.scheduler/dataSource.tests.js:765]: Question.
Why do we need this ```try catch``` block here?

**Review by williamvinogradov — DISMISSED** [review]

File: packages/devextreme/testing/helpers/scheduler/helpers.js (hunk 1 above)
Comment (Alyar666) [reply packages/devextreme/testing/helpers/scheduler/helpers.js:276]: Yes, you are right

File: packages/devextreme/testing/helpers/scheduler/helpers.js (hunk 2 above)
Comment (Alyar666) [reply packages/devextreme/testing/helpers/scheduler/helpers.js:275]: In this case, the following error will appear:
Снимок экрана 2024-03-06 в 13 24 41

I think Qunit coverage doesn't work with this syntax.

Comment (williamvinogradov) [reply packages/devextreme/testing/helpers/scheduler/helpers.js:275]: Yup, got it
So, let's leave it as is

**Review by williamvinogradov — DISMISSED** [review]

**Review by Raushen — APPROVED** [review]

**Review by williamvinogradov — APPROVED** [review]

**Review by alexanderPolosatov — APPROVED** [review]

**Pull request merged**